sanic/server.py

```diff
@@ -295,6 +309,29 @@ def data_received(self, data):
         except Exception:
             logger.exception("protocol.data_received")
 
+    # -------------------------------------------- #
+    # Python 3.7+
+    # -------------------------------------------- #
+
+    def get_buffer(self, sizehint=-1):
+        return self._buffer
+
+    def buffer_updated(self, nbytes: int) -> None:
+        data = self._buffer[:nbytes]
+        self._time = current_time()
+        self.recv_buffer += data
```

Comment:
Do you need to clear `self._buffer` here?

what happens if `buffer_updated()` is called a second time with a new nbytes, should it still read `[:nbytes]` from the start of the buffer?

Does it need to keep track of buffer position like this implementation?:
https://gist.github.com/1st1/1c606e5b83ef0e9c41faf21564d75ad7#file-get_buffer_bench-py-L27-L43

Reply:
I have lots more questions here, (does get_buffer() get called again after buffer_updated() and before buffer_updated() is called again?, should you clear `self._buffer` before returning it in `get_buffer()`) I should probably just test it locally to clarify to myself exactly how this sequence works.

Reply:
Ok, I understand it better now. Resolving this thread.